src/main/java/marquez/service/models/SourceTypes.java

```diff
@@ -0,0 +1,348 @@
+      @Nullable final List<Filesystem> filesystems) {
+    this.dbMap = toDbMap(dbs);
+    this.streamMap = toStreamMap(streams);
+    this.filesystemMap = toFilesystemMap(filesystems);
+  }
+
+  private Map<String, Db> toDbMap(@Nullable final List<Db> overrides) {
+    final Map<String, Db> dbMap = Maps.newHashMap();
+    DBS.forEach(
+        db -> {
+          dbMap.put(db.getType().toUpperCase(), db);
+        });
+    // Overrides
+    if (overrides != null) {
+      overrides.forEach(
+          override -> {
+            dbMap.put(override.getType().toUpperCase(), override);
+          });
+    }
+    return ImmutableMap.copyOf(dbMap);
+  }
+
+  private Map<String, Stream> toStreamMap(@Nullable final List<Stream> overrides) {
+    final Map<String, Stream> streamMap = Maps.newHashMap();
+    STREAMS.forEach(
+        stream -> {
+          streamMap.put(stream.getType().toUpperCase(), stream);
+        });
+    // Overrides
+    if (overrides != null) {
+      overrides.forEach(
+          override -> {
+            streamMap.put(override.getType().toUpperCase(), override);
+          });
+    }
+    return ImmutableMap.copyOf(streamMap);
+  }
+
+  private Map<String, Filesystem> toFilesystemMap(@Nullable final List<Filesystem> overrides) {
+    final Map<String, Filesystem> filesystemMap = Maps.newHashMap();
+    FILESYSTEMS.forEach(
+        filesystem -> {
+          filesystemMap.put(filesystem.getType().toUpperCase(), filesystem);
+        });
+    // Overrides
+    if (overrides != null) {
+      overrides.forEach(
+          override -> {
+            filesystemMap.put(override.getType().toUpperCase(), override);
+          });
+    }
+    return ImmutableMap.copyOf(filesystemMap);
+  }
+
+  public boolean exists(@NonNull String type, @NonNull SourceQualifier qualifier) {
+    checkNotBlank(type, "type must not be blank");
+    final String typeAsUpperCase = type.toUpperCase();
+    switch (qualifier) {
+      case DB:
+        return dbMap.containsKey(typeAsUpperCase);
+      case STREAM:
+        return streamMap.containsKey(typeAsUpperCase);
+      case FILESYSTEM:
+        return filesystemMap.containsKey(typeAsUpperCase);
+      default:
+        throw new AssertionError();
+    }
+  }
+
+  public boolean typeHasSupportFor(
+      @NonNull String type, @NonNull SourceQualifier qualifier, @NonNull String dataType) {
+    checkNotBlank(type, "type must not be blank");
+    checkNotBlank(dataType, "dataType must not be blank");
+    if (!exists(type, qualifier)) {
+      return NO_SUPPORT;
+    }
+
+    final String typeAsUpperCase = type.toUpperCase();
+    switch (qualifier) {
+      case DB:
+        return dbMap.containsKey(typeAsUpperCase)
+            ? dbMap.get(typeAsUpperCase).supports(dataType)
+            : NO_SUPPORT;
+      default:
+        // We currently only support data types for a database source.
+        return NO_SUPPORT;
+    }
+  }
+
+  private static final Db MYSQL =
```

Comment:
I would add a comment to clarify those are defaults that can be overridden in the conf